internal/docstore/driver/driver.go

```diff
@@ -71,3 +83,28 @@ type Mod struct {
 	FieldPath []string
 	Value     interface{}
 }
+
+// An ActionListError contains all the errors encountered from a call to RunActions,
+// and the positions of the corresponding actions.
+type ActionListError []struct {
+	Index int
+	Err   error
+}
+
+// TODO(jba): use xerrors formatting.
+func (e ActionListError) Error() string {
+	var s []string
+	for _, x := range e {
+		s = append(s, fmt.Sprintf("at %d: %v", x.Index, x.Err))
+	}
+	return strings.Join(s, "; ")
+}
+
+func (e ActionListError) Unwrap() error {
```

Comment:
I don't understand what "Unwrap" is for. I find it surprising that it returns `nil` if there's more than one error. What is this used for?

Reply:
OK, now I see it used above. Maybe "SingleError" or something? At least a docstring.

Reply:
This is for conformance with the new errors stuff. See https://tip.golang.org/pkg/errors/#Wrapper.